test/e2e/capi_test.go

```diff
@@ -213,40 +213,63 @@ var _ = Describe("Running the Cluster API E2E tests", func() {
 	if os.Getenv("USE_LOCAL_KIND_REGISTRY") != "true" {
 		Context("API Version Upgrade", func() {
 			var e2eConfigNoAddons *clusterctl.E2EConfig
+			BeforeEach(func() {
+				// Unset resource group and vnet env variables, since the upgrade test creates 2 clusters,
+				// and will result in both the clusters using the same vnet and resource group.
+				Expect(os.Unsetenv(AzureResourceGroup)).To(Succeed())
+				Expect(os.Unsetenv(AzureVNetName)).To(Succeed())
+
+				// Unset windows specific variables
+				Expect(os.Unsetenv("WINDOWS_WORKER_MACHINE_COUNT")).To(Succeed())
+
+				Expect(os.Setenv("K8S_FEATURE_GATES", "WindowsHostProcessContainers=true")).To(Succeed())
+
+				e2eConfigNoAddons = &clusterctl.E2EConfig{}
+				e2eConfig.ManagementClusterName = e2eConfigNoAddons.ManagementClusterName
+				e2eConfigNoAddons.Variables = e2eConfig.Variables
+				e2eConfigNoAddons.Intervals = e2eConfig.Intervals
+
+				providers := []clusterctl.ProviderConfig{}
+				for _, provider := range e2eConfig.Providers {
+					if provider.Type != "AddonProvider" {
+						providers = append(providers, provider)
+					}
+				}
+				e2eConfigNoAddons.Providers = providers
 
-			Context("upgrade from an old version of v1beta1 to current, and scale workload clusters created in the old version", func() {
-				BeforeEach(func() {
-					// Unset resource group and vnet env variables, since the upgrade test creates 2 clusters,
-					// and will result in both the clusters using the same vnet and resource group.
-					Expect(os.Unsetenv(AzureResourceGroup)).To(Succeed())
-					Expect(os.Unsetenv(AzureVNetName)).To(Succeed())
-
-					// Unset windows specific variables
-					Expect(os.Unsetenv("WINDOWS_WORKER_MACHINE_COUNT")).To(Succeed())
-
-					Expect(os.Setenv("K8S_FEATURE_GATES", "WindowsHostProcessContainers=true")).To(Succeed())
-
-					e2eConfigNoAddons = &clusterctl.E2EConfig{}
-					e2eConfig.ManagementClusterName = e2eConfigNoAddons.ManagementClusterName
-					e2eConfigNoAddons.Variables = e2eConfig.Variables
-					e2eConfigNoAddons.Intervals = e2eConfig.Intervals
-
-					providers := []clusterctl.ProviderConfig{}
-					for _, provider := range e2eConfig.Providers {
-						if provider.Type != "AddonProvider" {
-							providers = append(providers, provider)
-						}
+				images := []clusterctl.ContainerImage{}
+				for _, image := range e2eConfig.Images {
+					if !strings.Contains(image.Name, "cluster-api-helm") {
+						images = append(images, image)
 					}
-					e2eConfigNoAddons.Providers = providers
+				}
+				e2eConfigNoAddons.Images = images
+			})
 
-					images := []clusterctl.ContainerImage{}
-					for _, image := range e2eConfig.Images {
-						if !strings.Contains(image.Name, "cluster-api-helm") {
-							images = append(images, image)
-						}
+			Context("upgrade from an old version of v1beta1 to current, and scale workload clusters created in the old version", func() {
```

Comment:
I wonder if we could surround these in a for loop?

```go
	for _, s := range []struct{name string; variable string}{
	    {
	        "old",
	        e2eConfig.GetVariable(OldCAPIUpgradeVersion),
	    },
            {
                "latest",
                e2eConfig.GetVariable(LatestCAPIUpgradeVersion),
            },
	}
```

Seems like everything is the same for both scenarios.

Reply:
I owe you a 🍺 or equivalent gesture for trying that out.